### ISSUE TITLE
Testing Native App using Node.js

### ISSUE THREAD

**Issue by SGarno** [Microsoft/WinAppDriver#626]
I am trying to automate a native windows application using Node.js.    I keep getting that Locator Strategy is not supported for this session.    Does that mean the 'appium-windows-driver' node module is not able to work with WAD 1.1?   Or is there something wrong with my code.  

It seems to start notepad ok, but nothing seems to work and there really aren't any docs regarding using WAD with Node.js.    The node module I am using doesn't seem conform to the API docs for Java or C#  (i.e. there is no function FindElementByName, etc) either.   The only function I could find was **findElement**.   

However, nothing seems to work when I try to use it.    I have tried passing in 'id', 'name', etc and tried XPath.    I assume I am supposed to be using XPath, but again, I can't find anything out there that helps.

Has anyone been able to make this work from Node?

Here is the code I have currently:

```
let WindowsDriver = require('appium-windows-driver').WindowsDriver;

let defaultCaps = {
  platformName: 'Windows',
  deviceName: 'WindowsPC',
  app: 'C:\\Windows\\System32\\notepad.exe'
};

async function main() {
  let driver = new WindowsDriver();
  await driver.createSession(defaultCaps);
  let element = await driver.findElement('//*[@ClassName="Edit"]');
//  element.SendKeys("This is some text");
}

main();

```

**Comment by naeemakram** [comment]
Have you tried copying the output of Notepad sample as shown on the WAD console while executing a scenario?

**Comment by SGarno** [comment]
I am not sure I follow what you mean by copying the output?      The output from the WAD console is what I have provided above.   Am I missing something?

**Comment by naeemakram** [comment]
Aside from my last comment, I see the following in default capabilities. and deviceName is not in there. Maybe you could remove it. Just as an experiment. 

![image](https://user-images.githubusercontent.com/6885369/54222396-28a55900-4517-11e9-9146-aab82bf2db80.png)

**Comment by SGarno** [comment]
I think that is required...

Yep...get this error message:

(node:2116) UnhandledPromiseRejectionWarning: SessionNotCreatedError: A new session could not be created. Details: The desiredCapabilities object was not valid for the following reason(s): 'deviceName' can't be blank

**Comment by naeemakram** [comment]
At my home PC I'm doing same in C# and specifying only two capabilities, 

            appCapabilities.SetCapability(
                "app", @"C:\Windows\System32\notepad.exe");
            appCapabilities.SetCapability(
                "deviceName", "WindowsPC");

This seems to work. 

![image](https://user-images.githubusercontent.com/6885369/54229704-a886ef80-4526-11e9-81c0-ac40d88394f0.png)

**Comment by naeemakram** [comment]
I tried to send keys to the notedpad and it worked.
Some values are showing in the console output of the WinAppDriver, maybe they could help you. 
          ```
calcSession.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(1.5));

            var editBox = calcSession.FindElementByClassName("Edit");
            if(editBox == null)
            {
                Console.WriteLine("Edit not found.");
                return;
            }

            editBox.SendKeys("Hello World!");
```


![image](https://user-images.githubusercontent.com/6885369/54230102-8a6dbf00-4527-11e9-84b5-0652df102185.png)

**Comment by naeemakram** [comment]
![image](https://user-images.githubusercontent.com/6885369/54230187-b9843080-4527-11e9-9a60-cf90b25eea18.png)

**Comment by SGarno** [comment]
You are using C#, which seems to have the methods such as FindElementByClassName, whereas the JavaScript library doesn't have that.  This is where my problem seems to be.

**Comment by naeemakram** [comment]
Yes, I'm using C#. But you can achieve the same with Node.js by sending
GET/POST requests directly to the WinAppDriver as listed in the docs I
suppose.
API calls are after all just a wrapper on underlying JSON wire protocol.

**Comment by SGarno** [comment]
Yeah, I was trying to use the library rather than create my own API.    Despite the code being updated in February, it seems that Appium is not keeping up with the changes to WAD.    https://github.com/appium/appium-windows-driver

**Comment by yodurr** [comment]
Appium installs a specific version of WinAppDriver, and currently they install  v1.1
https://github.com/appium/appium-windows-driver/blob/master/lib/installer.js

We haven't submitted a PR yet to use 1.1.1 but can do so after we get more feedback on 1.1.1.
